arviz/plots/distcomparisonplot.py

```diff
@@ -134,6 +135,22 @@ def plot_dist_comparison(
         for data, var in zip(datasets, var_names)
     ]
 
+    total_plots = sum(
+        sum(1 for _ in xarray_var_iter(data, var_names=var, combined=True))
+        for data, var in zip(datasets, var_names)
+    )
+    maxplots = sum(len(splot) for splot in dc_plotters)
```

Comment:
`maxplots` as defined here is not necessarily equal to the rcParam, here plots are done in blocks of 3-4 subplots, so this will be equal _or lower_ than the rcParam. To avoid confusion, I think the warning should use the rcParam first, then in the second occurrence it can be `maxplots`. It is not necessary nor correct to loop over dc_plotters though, it should be:

    maxplots = len(dc_plotters[0]) *  (len(groups) + 1)